### ISSUE TITLE
Didn't find class "java.lang.invoke.MethodHandles$Lookup"

### ISSUE THREAD

**Issue by CraZyLegenD** [square/retrofit#3343]
//retrofit
 ```
   implementation 'com.squareup.retrofit2:converter-moshi:2.8.0'
   implementation 'com.squareup.retrofit2:retrofit:2.8.0'
   implementation 'com.squareup.retrofit2:adapter-rxjava2:2.8.0'
   implementation 'com.squareup.retrofit2:converter-gson:2.8.0'
```

It doesn't work with either moshi either gson converters.

Gradle configured with

```
compileOptions {
        sourceCompatibility = 1.8
        targetCompatibility = 1.8
    }

kotlinOptions {
        jvmTarget = "1.8"
    }
```

Also tried

```
compileOptions {
        sourceCompatibility JavaVersion.VERSION_1_8
        targetCompatibility JavaVersion.VERSION_1_8
    }
```

What's the error?

>  Caused by: java.lang.ClassNotFoundException: Didn't find class "java.lang.invoke.MethodHandles$Lookup" 

It happens only on:
Android 7.0
Android 7.1.1
Android 7.1.2

Where does it print the stack trace?

https://github.com/square/retrofit/blob/master/retrofit/src/main/java/retrofit2/Platform.java 
Line 62, 115, 44, 34, 37, 430

**Comment by CraZyLegenD** [comment]
Also worth mentioning that, the clean build didn't help, deleting the build folder inside my files didn't help.

**Comment by CraZyLegenD** [comment]
Tested from Android 5.0 till 11, had to double-check and confirm that it breaks on Android 7.X and 7.X.X releases

**Comment by drakeet** [comment]
The same error:
```
Caused by java.lang.NoClassDefFoundError: Failed resolution of: Ljava/lang/invoke/MethodHandles$Lookup;
       at retrofit2.Platform.<init> + 62(Platform.java:62)
       at retrofit2.Platform$Android.<init> + 115(Platform.java:115)
       at retrofit2.Platform.findPlatform + 44(Platform.java:44)
       at retrofit2.Platform.<clinit> + 34(Platform.java:34)
       at retrofit2.Platform.get + 37(Platform.java:37)
       at retrofit2.Retrofit$Builder.<init> + 430(Retrofit.java:430)
       at com.drakeet.purewriter.Retrofits.<clinit> + 45(Retrofits.java:45)
       at com.drakeet.purewriter.PureWriter$onCreate$1.run + 104(PureWriter.java:104)
       at io.reactivex.internal.operators.completable.CompletableFromAction.subscribeActual + 35(CompletableFromAction.java:35)
       at io.reactivex.Completable.subscribe + 2309(Completable.java:2309)
       at io.reactivex.internal.operators.completable.CompletableSubscribeOn$SubscribeOnObserver.run + 64(CompletableSubscribeOn.java:64)
       at io.reactivex.Scheduler$DisposeTask.run + 578(Scheduler.java:578)
       at io.reactivex.internal.schedulers.ScheduledRunnable.run + 66(ScheduledRunnable.java:66)
       at io.reactivex.internal.schedulers.ScheduledRunnable.call + 57(ScheduledRunnable.java:57)
       at java.util.concurrent.FutureTask.run + 237(FutureTask.java:237)
       at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run + 272(ScheduledThreadPoolExecutor.java:272)
       at java.util.concurrent.ThreadPoolExecutor.runWorker + 1133(ThreadPoolExecutor.java:1133)
       at java.util.concurrent.ThreadPoolExecutor$Worker.run + 607(ThreadPoolExecutor.java:607)
       at java.lang.Thread.run + 761(Thread.java:761)
```

**Comment by dimipaun** [comment]
I see the same on Android 7.1.1:
```
Fatal Exception: java.lang.NoClassDefFoundError: Failed resolution of: Ljava/lang/invoke/MethodHandles$Lookup;
       at retrofit2.Platform.<init>(Platform.java)
       at retrofit2.Platform$Android.<init>(Platform.java)
       at retrofit2.Platform.findPlatform(Platform.java)
       at retrofit2.Platform.<clinit>(Platform.java)
       at retrofit2.Platform.get(Platform.java)
       at retrofit2.Retrofit$Builder.<init>(Retrofit.java)
       at com.twitter.sdk.android.core.internal.oauth.OAuthService.<init>(OAuthService.java)
       at com.twitter.sdk.android.core.internal.oauth.OAuth2Service.<init>(OAuth2Service.java)
       at com.twitter.sdk.android.core.TwitterCore.createGuestSessionProvider(TwitterCore.java)
       at com.twitter.sdk.android.core.TwitterCore.getGuestSessionProvider(TwitterCore.java)
       at com.twitter.sdk.android.core.TwitterCore.addApiClient(TwitterCore.java)
       at com.twitter.sdk.android.core.TwitterCore.lambda$getInstance$0(TwitterCore.java)
       at com.twitter.sdk.android.core.-$$Lambda$TwitterCore$ltCfCYbT02qdRyGLE4y1z_Jg9Os.run(-.java)
       at java.util.concurrent.ThreadPoolExecutor.runWorker + 1133(ThreadPoolExecutor.java:1133)
       at java.util.concurrent.ThreadPoolExecutor$Worker.run + 607(ThreadPoolExecutor.java:607)
       at java.lang.Thread.run + 761(Thread.java:761)
```

**Comment by ashley-figueira** [comment]
What is the fix for this? I've upgraded to Retrofit 2.9.0 and im still getting this crash for Android 7.X ... Using Java 1.8 and gradle plugin 3.6.1

**Comment by CraZyLegenD** [comment]
> What is the fix for this? I've upgraded to Retrofit 2.9.0 and im still getting this crash for Android 7.X ... Using Java 1.8 and gradle plugin 3.6.1

Did you update your retrofit adapters too?

**Comment by ashley-figueira** [comment]
> > What is the fix for this? I've upgraded to Retrofit 2.9.0 and im still getting this crash for Android 7.X ... Using Java 1.8 and gradle plugin 3.6.1
> 
> Did you update your retrofit adapters too?

I did.

**Comment by CraZyLegenD** [comment]
> > > What is the fix for this? I've upgraded to Retrofit 2.9.0 and im still getting this crash for Android 7.X ... Using Java 1.8 and gradle plugin 3.6.1
> > 
> > 
> > Did you update your retrofit adapters too?
> 
> I did.

I just tested it on Android 7.1.1

```
def retrofit "2.9.0
   implementation "com.squareup.retrofit2:converter-moshi:$retrofit"
    implementation "com.squareup.retrofit2:retrofit:$retrofit"
    implementation "com.squareup.retrofit2:adapter-rxjava3:$retrofit"
    implementation "com.squareup.retrofit2:converter-gson:$retrofit"
    implementation 'com.squareup.okhttp3:logging-interceptor:4.7.2'
```

I didn't get any crash
